src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryImpl.java

```diff
@@ -165,9 +166,28 @@ public Optional<byte[]> getResourceContent( String url ) throws IOException
             transporter.get( task );
             return Optional.of( task.getDataBytes() );
         }
+        catch ( ResourceDoesNotExistException e )
+        {
+            if ( LOGGER.isDebugEnabled() )
+            {
+                LOGGER.debug( "Cache item not found: {}", getFullUrl( url ), e );
+            }
+            else
+            {
+                LOGGER.info( "Cache item not found: {}", getFullUrl( url ) );
+            }
+            return Optional.empty();
+        }
         catch ( Exception e )
         {
-            LOGGER.info( "Cannot download {}", getFullUrl( url ), e );
+            if ( LOGGER.isDebugEnabled() )
+            {
+                LOGGER.debug( "Cannot download cache item {}", getFullUrl( url ), e );
+            }
+            else
+            {
+                LOGGER.error( "Cannot download cache item {}: {}", getFullUrl( url ), e );
+            }
```

Comment:
> Can I suggest that if we get here it should be at least a warning regardless of whether debug is enabled. I'd go for error() as the remote cache can be easily disabled if it is known that it's not available.

not sure to understand. if the remote cache is disabled. no download will be tried.

Reply:
Yes that's the scenario I had in mind, i.e. enabling the remote cache is a deliberate act so if it then doesn't work that should be an error not a warning.
In the situation where the user doesn't expect the remote cache to work, e.g. working offline, they should disable it.

Reply:
> i.e. enabling the remote cache is a deliberate act so if it then doesn't work that should be an error not a warning.
In the situation where the user doesn't expect the remote cache to work, e.g. working offline, they should disable it.

I don't think that's the intended scenario described by the author.

The remote cache can be working 100% correctly and still not contain the requested items. In fact, this is 100% bound to happen, unless the cache is magically warmed-up beforehand.

1. New project/branch/version/whatever so cache is empty.
2. Build starts
3. Cache extension cannot find cache item of Module X. <-- this is 100% normal and expected behavior and shouldn't throw an error (or even a warming i.m.h.o.), let alone a stack-trace.
4. Module X gets built
5. Module X output gets put into the cache
6. Next build starts, and now it can find cache item of Module X.

Having the logs filled with error/stack-traces for non-issues is counterproductive. So I support any attempt at a more sane output.

Reply:
@alexthomasmc Is this the remote cache "not working", though, or can this also just represent a cache miss / expired cache?

Cache misses certainly are not errors, heck they're not even problems, they're a standard aspect of how caching works. Data stored in a cache can be wiped out at any moment, for any reason (or even no reason at all). Caches go missing all the time.

When a build system fails to obtain a cached artifact, it simply needs to generate it anew. It can do that, it has the technology.

Reply:
The cache miss exception is caught just before this, so by the time we get here there's a bigger problem. Agree with your policy on this - I think that's the intent of the current logic, it's just the logging that's the issue.